adapter/internal/discovery/xds/server.go

```diff
@@ -802,6 +804,11 @@ func GenerateEnvoyResoucesForLabel(label string) ([]types.Resource, []types.Reso
 				}
 				clusterArray = append(clusterArray, orgIDOpenAPIClustersMap[organizationID][apiKey]...)
 				vhostToRouteArrayMap[vhost] = append(vhostToRouteArrayMap[vhost], orgIDOpenAPIRoutesMap[organizationID][apiKey]...)
+				if arrayContains(knownVhostPrefixes, strings.Split(vhost, ".")[0]) {
```

Comment:
Shall we keep some sort of a variable to decide on whether we need to add two routes. If some issue occurs later, then changing it via a environment property would be the fastest.

Reply:
fixed via https://github.com/wso2/product-microgateway/pull/3401/commits/fc514e366269a7d18753687cdedd95c0a768afec